File: prog2.c
#include<stdio.h>
#include<math.h>

int main()
{
    int binary;
    printf("Enter the binary number: ");
    scanf("%i",&binary);
    int sum=0;
    double i=0;

    while (binary>0)
    {
        int rem=binary%10;
        binary=binary/10;

        double power=pow(2.0,i);
        sum+=rem*(int)power;
        i++;
    }
    printf("%i",sum);

    return 0;
}

File: prog2.py
bin=int(input("Enter the binary value: "))
bin2=bin
dec=0
i=0
flag=0

while bin>0:
    rem=bin%10
    if rem!=0 and rem!=1:
        print(f'{bin2} binary value not supported')
        flag=1
        break
    bin=bin//10
    dec+=(rem*(2**i))
    i+=1

if flag==0:
    print(f'Decimal conversion for {bin2} is: {dec}')